### PR TITLE
Try mocha/chai test runners

### PR DESCRIPTION
**Pull request by drwpow**
## Changes

In switching to Vite and having to rework tests with the new compiler, I took the opportunity to revisit our test runner. Since Jest had just released ESM support, it seemed like a good option. The biggest motivator for switching from uvu -> Jest is simply contributor familiarity; I wanted to prioritize reducing the barrier to contribute testing using what most JS devs are familiar with. But other benefits were timeouts, clearer errors, more utilities, and an overall more stable runner. Fortunately Jest worked! 🎉  … But only locally 😑. It has [not been passing CI](https://github.com/snowpackjs/astro/pull/1406/checks).

This PR switches to Mocha / Chai, again with the priority of “use what most people know.” This is the closest thing to Jest, and if you know Jest you probably can pick this up quickly.

Basically:

**Before**

Screen Shot 2021-09-23 at 14 53 18

**After**

Screen Shot 2021-09-23 at 14 53 15

## Testing

Testing PR.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -97,45 +97,47 @@ jobs:
...
-        run: yarn build:all
-
-      - name: "Smoke Test: Build 'docs'"
-        run: yarn build
-        working-directory: ./docs
-
-      - name: "Smoke Test: Build 'www'"
-        run: yarn build
-        working-directory: ./www
+  # NOTE: temporarily disabled until `next` branch can build docs again
```
Comment (drwpow) [inline .github/workflows/ci.yml:100]: Temporarily disabled docs build test until we get a little farther

File: packages/astro/src/internal/index.ts [hunk 2]
```diff
@@ -77,11 +76,11 @@ export interface AstroComponentFactory {
   isAstroComponentFactory?: boolean;
 }
 
-export const createComponent = (cb: AstroComponentFactory) => {
+export function createComponent(cb: AstroComponentFactory) {
```
Comment (drwpow) [inline packages/astro/src/internal/index.ts:79]: Minor code style nit: I think functions are generally better because they all get hoisted and don’t depend on order of definition like `const` does

File: packages/astro/src/runtime/astro.ts [hunk 3]
```diff
@@ -1,41 +0,0 @@
-import type { AstroComponent, AstroComponentFactory } from '../internal';
-
-import { spreadAttributes, defineStyleVars, defineScriptVars } from '../internal';
```
Comment (drwpow) [inline packages/astro/src/runtime/astro.ts:3]: This file was deleted because this introduced a circular dependency between `src/runtime/astro.ts` and `src/internal/index.ts`. This was just moved into `internal`.

Naming is hard.

File: packages/astro/src/runtime/ssr.ts [hunk 4]
```diff
@@ -87,8 +88,8 @@ async function resolveRenderers(viteServer: ViteDevServer, ids: string[]) {
   return renderers;
 }
 
-async function resolveImportedModules(viteServer: ViteDevServer, file: string) {
-  const { url } = await viteServer.moduleGraph.ensureEntryFromUrl(file);
+async function resolveImportedModules(viteServer: ViteDevServer, file: URL) {
+  const { url } = await viteServer.moduleGraph.ensureEntryFromUrl(slash(fileURLToPath(file))); // note: for some reason Vite expects forward slashes here for Windows, which `slash()` helps resolve
```
Comment (drwpow) [inline packages/astro/src/runtime/ssr.ts:92]: Fun Windows Vite bug! 🎉 New test runner caught it.

File: packages/astro/test/astro-attrs.test.js [hunk 5]
```diff
@@ -25,32 +26,32 @@ describe('Attributes', () => {
...
-  test('Passes boolean attributes to components as expected', async () => {
+  it('Passes boolean attributes to components as expected', async () => {
     const html = await fixture.readFile('/component/index.html');
     const $ = cheerio.load(html);
 
-    expect($('#true').attr('attr')).toBe('attr-true');
-    expect($('#true').attr('type')).toBe('boolean');
-    expect($('#false').attr('attr')).toBe('attr-false');
-    expect($('#false').attr('type')).toBe('boolean');
+    expect($('#true').attr('attr')).to.equal('attr-true');
```
Comment (drwpow) [inline packages/astro/test/astro-attrs.test.js:37]: Mocha + Chai was selected because of its similarity to Jest. Migrating was just a quick find-and-replace.

**Comment by FredKSchott** [comment]
AMAZING, nice work

**Comment by jasikpark** [comment]
Nice! I've had a great experience w/ Ava in https://github.com/snowpackjs/prettier-plugin-astro - it's fully ESM in Avajs 4.0 & has a similar setup to uvu, but is much more reliable!

https://github.com/avajs/ava
